src/lib/transaction_logic/test/zkapp_cmd_builder.ml

```diff
@@ -31,27 +31,35 @@ let get_nonce_exn (pk : Public_key.Compressed.t) :
   in
   nonce
 
-let update_body ?(update = Account_update.Update.noop) ~account amount =
+let update_body ?preconditions ?(update = Account_update.Update.noop) ~account
+    amount =
   let open Monad_lib.State.Let_syntax in
   let open Account_update in
-  let%map nonce = get_nonce_exn account in
+  let account_id = Test_account.account_id account in
+  let%map default =
+    let%map nonce = get_nonce_exn account.pk in
+    Account_update.Preconditions.
+      { network = Zkapp_precondition.Protocol_state.accept
+      ; account = Account_precondition.Nonce nonce
+      ; valid_while = Ignore
+      }
+  in
+  let update_preconditions = Option.value ~default preconditions in
   let account_update = update in
+  let tok_id = Account_id.token_id account_id in
   Body.
     { dummy with
-      public_key = account
+      public_key = Account_id.public_key account_id
     ; update = account_update
-    ; token_id = Token_id.default
+    ; token_id = tok_id
     ; balance_change = amount
     ; increment_nonce = true
     ; implicit_account_creation_fee = true
-    ; may_use_token = No
+    ; may_use_token =
+        (if Token_id.(equal default tok_id) then No else Parents_own_token)
     ; authorization_kind = Signature
     ; use_full_commitment = true
-    ; preconditions =
-        { network = Zkapp_precondition.Protocol_state.accept
-        ; account = Account_precondition.Nonce nonce
-        ; valid_while = Ignore
-        }
+    ; preconditions = update_preconditions
```

Comment:
```suggestion
    ; preconditions
```

Reply:
The same problem, `preconditions` gets shadowed.